projects/vs/psa-boot-test/psa-boot-test.cpp

```diff
@@ -5,72 +5,222 @@
+        /// <summary>
+        /// In this case, we use an invalid package 
+        /// that has a wrong size.
+        /// The decrypted image must not be the same as 
+        /// the expected plain image value. 
+        /// </summary>
+        TEST_METHOD(decryption_test_wrong_size)
+        {
+            boot_upgrade_package_t* package = GET_TEST_PACKAGE(td_package_wrong_size);
+            {
+                bool isEqual = arrays_are_equal(package->image, td_valid_package_plain_image_content, td_valid_package_plain_image_len);
+                /* We should have encrypted image so, strings should not be the same before decryption */
+                Assert::IsFalse(isEqual);
+                /* Let us decrypt the valid package */
+                boot_decrypt_upgrade_package(package);
+                isEqual = arrays_are_equal(package->image, td_valid_package_plain_image_content, td_valid_package_plain_image_len);
+                /* After the encryption, the image must not be the same with expected plain image */
+                Assert::IsFalse(isEqual);
+            }
+            RELEASE_TEST_PACKAGE(package);
+        }
+        /// <summary>
+        /// In this case, we use an invalid package 
+        /// that has an invalid signature.
+        /// The image must not be authenticated 
+        /// using the signing public key. 
+        /// </summary
+        TEST_METHOD(authenticate_test_invalid_signature)
+        {
+            boot_upgrade_package_t* package = GET_TEST_PACKAGE(td_package_invalid_signature);
+            {                
+                /* Let us the valid package */
+                boot_authenticate_upgrade_package(package);
+               bool isEqual = arrays_are_equal(package->metadata.signature, td_valid_package_plain_image_content, td_valid_package_plain_image_len);
+                /* After the encryption, the image must not be the same with expected plain image */
+                Assert::IsFalse(isEqual);
+            }
+            RELEASE_TEST_PACKAGE(package);
+        }
+        /// <summary>
+        /// In this case, we use an invalid package 
+        /// that has a wrong size.
+        /// The image must not be authenticated 
+        /// using the signing public key.
+        /// </summary>
+        TEST_METHOD(authentication_test_wrong_size)
+        {
+            boot_upgrade_package_t* package = GET_TEST_PACKAGE(td_package_wrong_size);
+            {
+                bool isEqual = arrays_are_equal(package->image, td_valid_package_plain_image_content, td_valid_package_plain_image_len);
+                /* We should have encrypted image so, strings should not be the same before decryption */
+                Assert::IsFalse(isEqual);
+                /* Let us decrypt the valid package */
+                boot_authenticate_upgrade_package(package);
+                isEqual = arrays_are_equal(package->image, td_valid_package_plain_image_content, td_valid_package_plain_image_len);
```

Comment:
You dont get and check the return value of boot_authenticate_upgrade_package().
You need to check only the return value here like authenticate_test_valid_image() test case above.